pkg/settings/setting.go

```diff
@@ -96,7 +96,7 @@ var (
 	WinsAgentVersion                    = NewSetting("wins-agent-version", "")
 	CSIProxyAgentVersion                = NewSetting("csi-proxy-agent-version", "")
 	CSIProxyAgentURL                    = NewSetting("csi-proxy-agent-url", "https://acs-mirror.azureedge.net/csi-proxy/%[1]s/binaries/csi-proxy-%[1]s.tar.gz")
-	SystemAgentInstallScript            = NewSetting("system-agent-install-script", "") // Defined via environment variable
+	SystemAgentInstallScript            = NewSetting("system-agent-install-script", "https://github.com/rancher/system-agent/releases/download/v0.3.5-rc1/install.sh") // Defined via environment variable
```

Comment:
I've been giving this some thought.

Rather than define an identical value here as a default, perhaps we should do this
1. Rename the `ENV CATTLE_SYSTEM_AGENT_INSTALL_SCRIPT` to something like `ENV DEFAULT_CATTLE_SYSTEM_AGENT_INSTALL_SCRIPT` and update subsequent references to it
2. Change this to `os.Getenv("DEFAULT_CATTLE_SYSTEM_AGENT_INSTALL_SCRIPT")`

This will bring the business logic back into the realm of doing what it is supposed to, i.e. if the system agent install script is NOT the default value, retrieve from URL, otherwise trust what is on disk.

The ideal fix would actually be to properly inject the system-agent-install-script default setting value at build time, but that is a can of worms I don't think is worth trying to open right now.

Reply:
If we want to stick with this approach, then please expand upon the comment here. You should make sure that we have behavior built into Rancher that relies upon the default value being hardcoded.

Reply:
@Oats87 will PR https://github.com/rancher/rancher/pull/44687 suffice as a fix?

Reply:
Updated the comment.